app/packages/app/src/Root/Datasets/Dataset.tsx

```diff
@@ -97,17 +98,24 @@ export const Dataset: Route<DatasetQuery> = ({ prepared }) => {
   const update = useStateUpdate();
 
   useEffect(() => {
-    update(({ reset }) => {
+    update(({ reset, get }) => {
       reset(filters);
-      reset(_activeFields({ modal: false }));
       reset(similarityParameters);
 
+      const newDataset = transformDataset(dataset);
+
+      const oldDataset = get(atoms.dataset);
+      oldDataset && console.log(oldDataset.id, newDataset.id);
+      if (!oldDataset || oldDataset.id !== newDataset.id) {
+        reset(_activeFields({ modal: false }));
```

Comment:
Is there a reason this isn't in `useStateUpdate()`? Looks like there are similar things there.

Reply:
Good point, fixed